src/views/IdentityOverview/IdentityOverview.tsx

```diff
@@ -45,6 +45,22 @@ export function IdentityOverview({ identity }: Props): JSX.Element | null {
   const showDownloadPrompt =
     credentials && credentials.some(({ isDownloaded }) => !isDownloaded);
 
+  const [hasLegacyDid, setHasLegacyDid] = useState(false);
+
+  useEffect(() => {
+    (async () => {
+      if (await needLegacyDidCrypto(identity.did)) {
+        setHasLegacyDid(true);
+      } else {
+        setHasLegacyDid(false);
+      }
+    })();
+  }, [identity, hasLegacyDid]);
```

Comment:
Does it really depend on hasLegacyDid?
```suggestion
  }, [identity]);
```